### PR TITLE
RootFinders: avoid stream usage in header

### PR DESCRIPTION
**Pull request by akva2**
put logging functions in separate compile unit where we use {fmt}

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: src/opm/common/utility/numeric/RootFinders.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+
+double ThrowOnError::handleBracketingFailure(const double x0, const double x1,
+                                             const double f0, const double f1)
+{
+    const auto str = fmt::format("Error in parameters, zero not bracketed: "
+                                 "[a, b] = [{}, {}]    f(a) = {}    f(b) = {}",
+                                 x0, x1, f0, f1);
+    OpmLog::debug(str);
+    OPM_THROW_NOLOG(std::runtime_error, str);
```
Comment (bska) [inline src/opm/common/utility/numeric/RootFinders.cpp:38]: It's probably fine to depend on `RootFinders.hpp` including `ErrorMacros.hpp` which in turn includes `<stdexcept>`, but for the sake of making this code a bit more independent we should probably include `<stdexcept>` explicitly here.

On the other hand, it's not entirely clear to me that following this work `RootFinders.hpp` **needs** to include `ErrorMacros.hpp`.

**Review by bska — APPROVED** [review]
This looks good to me.  The only thing we might consider changing is the set header being brought in transitively here.

File: src/opm/common/utility/numeric/RootFinders.cpp (hunk 1 above)
Comment (akva2) [reply src/opm/common/utility/numeric/RootFinders.cpp:38]: it does, there's an OPM_THROW left further down. But I can include my deps in the source, no problem.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**